internal/injector/lineinfo/annotation.go

```diff
@@ -0,0 +1,160 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+package lineinfo
+
+import (
+	"bytes"
+	"go/ast"
+	"go/token"
+	"strconv"
+
+	"github.com/dave/dst"
+	"github.com/dave/dst/decorator"
+)
+
+const generated = "<generated>"
+
+type (
+	// annotationVisitor is an ast.Visitor that adds `//line` directives to the visited nodes to
+	// adjust their logical source location so it matches those of the original `*ast.File` tree.
```

Comment:
I thought is was done by dave/dst automatically. Is this comment missing some context?

Reply:
No, `dave/dst` does none of this...